src/documents/tasks.py

```diff
@@ -89,42 +90,28 @@ def train_classifier():
 
 @shared_task
 def consume_file(
-    path,
-    override_filename=None,
-    override_title=None,
-    override_correspondent_id=None,
-    override_document_type_id=None,
-    override_tag_ids=None,
-    task_id=None,
-    override_created=None,
-    override_owner_id=None,
-    override_archive_serial_num: Optional[int] = None,
+    input_doc: Dict,
```

Comment:
@stumpylog I'd assume this here is because parameters for shared tasks need to be JSON serializable.

Maybe you can look into other serializers for celery
* such as pickle https://docs.celeryq.dev/en/stable/userguide/calling.html#calling-serializers

Then there would be no need to transform data class instances into this JSON serializable format, you could use them as types in the method signature, and probably don't need those as_dict / from_dict methods in the data classes.

Reply:
I did think about pickle when originally changing to celery.  I wasn't sure about the security implications.  The Redis broker _should_ be secured or otherwise not accessible, but I was concerned about it as a potentially untrusted source of data to unpickle.  And I didn't want to re-implement some sort of signing like `django-q` used.

It would remove the dataclass -> dict[str] conversion, and might be faster/smaller tasks in the broker.

Thoughts on the security?

Reply:
#2861 
I think there isn't a lot of security risk with using pickle.  The Redis instance should be secured and we suggest that in several places in the docs